packedForest/src/forestTypes/basicForests/rfClassification/fpForestClassificationBase.h

```diff
@@ -169,18 +169,115 @@ inline int predictClass(const T* observation){
 			}
 
 			inline float reportOOB(){
-				// for tree_i in trees sum OOB_i and normalize by the number of trees.
-				float oobi = 0;
-				int treesWithOOB = 0;
-				for (unsigned int i = 0; i < trees.size(); ++i){
-					if (trees[i].returnTotalOOB() > 0) {
-					  oobi += trees[i].returnOOB();
-					  treesWithOOB++;
+				int numCorrect = 0;
+				int numOOB = 0;
+				float oobAccuracy = 0;
+
+				std::map<int, int> oobBestClass;
+				// A vector of vectors (numObs X numClasses) for storing
+				// the class tallies.
+			    std::vector<std::vector<int>> oobClassVotes(fpSingleton::getSingleton().returnNumObservations(), std::vector<int>(fpSingleton::getSingleton().returnNumClasses(), 0));
+
+				// Iterate over trees to get oob points and add up their
+				// class votes.
+				for (unsigned int i = 0; i < trees.size(); i++){
```

Comment:
could just iterate the trees instead of an index into each tree. maybe cleaner that way?